src/pages/learn.js

```diff
@@ -1,25 +1,48 @@
-import React, { useContext } from "react"
+import React, { useContext, useEffect } from "react"
 import Navbar from "../components/navbar"
 import IDE from "../components/ide"
 import Term from "../components/term"
 import Video from "../components/video"
-import { HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
+import { HeartwoodStateContext, HeartwoodDispatchContext } from "../state/HeartwoodContextProvider"
 
 import "../styles/video.css"
 
-export default () => {
-  const dispatch = useContext(HeartwoodDispatchContext)
+const Learn = ({ location }) => {
+  const state = useContext(HeartwoodStateContext)
+
+  // Jacob, make sure to turn this back to || false when you are done testing
+  const { showPage } = location.state || false
+
+  useEffect(() => {
+    const gatsby = require("gatsby")
+    if (!showPage) {
+      gatsby.navigate("/")
+    }
+  }, [])
```

Comment:
Codacy found an issue: [React Hook useEffect has a missing dependency: 'showPage'. Either include it or remove the dependency array.](https://app.codacy.com/gh/treetoplearning/heartwood/pullRequest?prid=5837031)